fundamentals/impl.tex

```diff
@@ -392,7 +392,51 @@ \subsubsection{Resources and Materials}
 
 \subsubsection{Dynamic Resource Exchange}
 
-\TODO{Matt Gidden, please fill in}
+Dynamic Resource Exchange (DRE) is the critical resource-related process on
+which \Cyclus simulations are built. It defines the resource interaction
+mechanisms and methodologies for all agents. As the \Cyclus supply-demand
+resolution framework, the DRE must be able to support arbitrary
+facilities. Supporting the general \Cyclus philosophy, facilities are treated as
+black boxes and a clear supply-demand communication framework is defined.
+
+The DRE consists of three steps: supply-demand information gathering, resource
```

Comment:
This paragraph is very long. I would recommend separating it very clearly into : 

- One paragraph that names the three steps
- For each of those steps, a separate paragraph laying out the example for each step

(so, 4 paragraphs.) What do you think?